Stratis.Bitcoin.Features.Wallet/WalletSyncManager.cs

```diff
@@ -88,44 +87,47 @@ public virtual void ProcessBlock(Block block)
                     while (this.chain.GetBlock(fork.HashBlock) == null)
                         fork = fork.Previous;
 
-                    Guard.Assert(fork.HashBlock == block.Header.HashPrevBlock);
                     this.walletManager.RemoveBlocks(fork);
                 }
-                else
+
+                ChainedBlock incomingBlock = this.chain.GetBlock(block.GetHash());
+                if (incomingBlock.Height > this.walletTip.Height)
                 {
-                    ChainedBlock incomingBlock = this.chain.GetBlock(block.GetHash());
-                    if (incomingBlock.Height > this.walletTip.Height)
+                    // the wallet is falling behind we need to catch up
+                    var next = this.walletTip;
+                    while(next != incomingBlock)
                     {
-                        // the wallet is falling behind we need to catch up
-                        var next = this.walletTip;
-                        while(next != incomingBlock)
+                        // while the wallet is catching up the entire node will wait
```

Comment:
Please use capital letters at the beginning and full stops at the end of sentences :-)